File: backend/src/main/java/com/programmers/pcquotation/domain/member/entitiy/Member.java
package com.programmers.pcquotation.domain.member.entitiy;

import java.util.Collection;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;


public interface Member<T> {
	long getId();
	String getUsername();
	Collection<? extends GrantedAuthority> getAuthorities();
	String getApiKey();
	String getPassword();
}

File: backend/src/main/java/com/programmers/pcquotation/domain/member/entitiy/Member.kt
package com.programmers.pcquotation.domain.member.entitiy

import org.springframework.security.core.GrantedAuthority

interface Member<T> {
    fun getId():Long?
    fun getUsername():String?
    fun getAuthorities():Collection<GrantedAuthority>?
    fun getApiKey():String
    fun getPassword():String
}